doc/subsystems/conf_s11n/conf_s11n.rst

```diff
@@ -0,0 +1,156 @@
+.. _conf_s11n
+
+Runtime configuration system
+############################
+
+Config subsystem is meant for persisting per-device configuration
+and runtime state for modules.
+
+Configuration items are stored as key-value pairs, where both the key and
+the value are expected to be strings. Keys are divided into component
+elements, where packages register their subtree using the first element.
+E.g key ``id/serial`` consists of 2 components, ``id`` and ``serial``.
+Package sys/id registered it's subtree of configuration elements to be
+under ``id``.
+
+There are convenience routines for converting value back and forth
+from string.
+
+Handlers
+********
+
+Config handlers for subtree implement a set of handler functions.
+These are registered using a call to ``conf_register()``.
+
+**ch_get**
+    This gets called when somebody asks for a config element value
+    by it's name using ``conf_get_value()``.
+
+**ch_set**
+    This is called when value is being set using ``conf_set_value()``, and
+    also when configuration is loaded from persisted storage with
+    ``conf_load()``.
+
+**ch_commit**
+    This gets called after configuration has been loaded in full.
+    Sometimes you don't want individual configuration value to take
+    effect right away, for example if there are multiple settings
+    which are interdependent.
+
+**ch_export**
+    This gets called to dump all current configuration. This happens
+    when ``conf_save()`` tries to save the settings or transfer to any
+    user-implemented back-end.
+
+Persistence
+***********
+
+Backend storage for the config can be either FCB, a file in filesystem,
+or both.
+
+You can declare multiple sources for configuration; settings from
+all of these are restored when ``conf_load()`` is called.
+
+There can be only one target for writing configuration; this is where
+data is stored when you call ``conf_save()``, or conf_save_one().
+
+FCB read target is registered using ``conf_fcb_src()``, and write target
+using ``conf_fcb_dst()``. ``conf_fcb_src()`` as side-effect initializes the
+FCB area, so it must be called when calling ``conf_fcb_dst()``.
+File read target is registered using ``conf_file_src()``, and write target
+``conf_file_dst()``.
+
+Example: Device Configuration
+*****************************
+
+This is a simple example, config handler only implements ``ch_set`` and
+``ch_export``. ``ch_set`` is called when value is restored from storage (or
+when set initially), and ``ch_export`` is used to write the value to
+storage thanks to ''storage_func'' (it is possible to implement another export
```

Comment:
``storage_func()``.  The user can also implement some other export
    functionality, for example, writing to the shell console).